### DISCUSSION TITLE
about MiscValuesEpoch error

### DISCUSSION THREAD

**Discussion by zdwangjian5** [groops-devs/groops#279] [Q&A] [answered]
After i write some values into file with the type of MISCVALUES, how to read? when i use instrument::read，the type is MISCVALUES, "(std::unique_ptr<Epoch>(Epoch::create(type))" will get an error of "unknown instrument type (-1)" due to the unknown size of MISCVALUES, shouldnot anywhere to decode "# MISCVALUES(2)" to get the value sizes?

**Comment by tmayerguerr** [comment, marked as answer]
Hi,
I don't know how you created this file. But MISCVALUES has the number of data columns as its type. 
In the source code, a MiscValuesEpoch is initialized with the number of columns.
```
  MiscValuesArc arc;
  MiscValuesEpoch epoch(count);
  epoch.values(0) = ...;
  epoch.values(1) = ...;
  epoch.values(2) = ...;
   ...
  arc.push_back(epoch);
  writeFileInstrument(fileNameOut, arc);
```
Hope this helps
regards
Torsten

**Reply by zdwangjian5** [reply]
Thank you!
